packages/shared/src/components/squads/PublicSquadEligibilityCard.tsx

```diff
@@ -10,42 +16,46 @@ import { useLazyModal } from '../../hooks/useLazyModal';
 import { LazyModal } from '../modals/common/types';
 import { anchorDefaultRel } from '../../lib/strings';
 import {
-  SquadPostsProgressBar,
   MIN_SQUAD_POSTS,
+  SquadPostsProgressBar,
 } from './SquadPostsProgressBar';
 import { squadsPublicGuide } from '../../lib/constants';
+import { SquadStatus } from './settings';
 
-interface Props {
-  postsCount: number;
-  inReview: boolean;
-  squadId: string;
+interface PublicSquadEligibilityCardProps {
+  placeholder: ReactNode;
 }
 
-const PublicSquadEligibilityCard = ({
-  postsCount,
-  inReview,
-  squadId,
-}: Props): ReactElement => {
+export function PublicSquadEligibilityCard({
+  placeholder,
+}: PublicSquadEligibilityCardProps): ReactElement {
+  const router = useRouter();
+  const { squad } = useSquad({ handle: router.query.handle as string });
+  const { isFetched, status } = usePublicSquadRequests({
+    sourceId: squad?.id,
+  });
+  const postsCount = squad?.flags.totalPosts;
   const { shouldUseMobileFeedLayout } = useFeedLayout();
-  const { completeAction, checkHasCompleted, isActionsFetched } = useActions();
+  const { completeAction } = useActions();
   const { openModal } = useLazyModal();
-  const hideCard = useMemo(
-    () =>
-      isActionsFetched &&
-      checkHasCompleted(ActionType.HidePublicSquadEligibilityCard),
-    [checkHasCompleted, isActionsFetched],
-  );
 
   const onDismiss = useCallback(() => {
     completeAction(ActionType.HidePublicSquadEligibilityCard);
   }, [completeAction]);
 
   const onSubmit = useCallback(() => {
-    openModal({ type: LazyModal.SubmitSquadForReview, props: { squadId } });
-  }, [openModal, squadId]);
+    if (!squad.id) {
+      return;
+    }
 
-  if (hideCard) {
```

Comment:
We can't return `null` from the card as it would affect the actual placement vs the virtual placement of the card (posts after this would be pushed 1 item backwards).

Reply:
Have you checked what we do with other cards like user acquisition?

Reply:
User acquisition is also generated from the same hook, but not sure if we return null from there.

Reply:
with that we actually replace the first ad with the card. I think at some point we decided to do the same here to not deal with post shift. Let me know if I'm wrong though about the decisions for it

Reply:
we seem to be doing that as well here just saw the last file so all good